src/components/Wallet/ConnectWallet.tsx

```diff
@@ -0,0 +1,136 @@
+import { useAccount, useBalance, useConnect, useDisconnect } from "wagmi";
+import { Button } from "../Button";
+import { Popover, PopoverButton, PopoverPanel } from "@headlessui/react";
+import { InkIcon } from "../..";
+import { Address } from "viem";
+import { classNames } from "../../util/classes";
+import { trimAddress } from "../../util/trim";
+import { inkSepolia } from "wagmi/chains";
+import { useEnsImageOrDefault } from "../../hooks/useEnsImageOrDefault";
+import { useEnsNameOrDefault } from "../../hooks/useEnsNameOrDefault";
+import { InternalButton } from "../Button/InternalButton";
+
+export interface ConnectWalletProps {
+  className?: string;
+}
+
+export const ConnectWallet: React.FC<ConnectWalletProps> = ({ className }) => {
+  const { address, isConnected } = useAccount();
+  const { connect, connectors } = useConnect();
+  const ensName = useEnsNameOrDefault({ address });
+  const ensImage = useEnsImageOrDefault({ address });
```

Comment:
I'm out of time tonight so I'll skim over this quickly, but we need to make sure whatever we use to connect is the lightest and fastest option possible.